lxd/db/cluster/entities.go

```diff
@@ -733,3 +733,240 @@ func GetEntityURLs(ctx context.Context, tx *sql.Tx, projectName string, entityTy
 
 	return result, nil
 }
+
+/*
+The following queries return the ID of an entity by the information contained in its unique URL in a common format. This
+allows us to query for the IDs of multiple entities at once by concatenating these queries with UNION.
+All of the below queries expect as arguments the project name query parameter, the location query parameter, and the path arguments of the URL.
+Each row returned by all of these queries has two columns:
+ 1. An identifier for the query being executed, this is directly returned so that we can match the result of the query
+    to the URL (see PopulateEntityReferencesFromURLs below).
+ 2. The ID of the entity.
+
+Note: It may be possible to further improve the efficiency of these queries by using WHERE statements, e.g. `WHERE projects.name IN (?, ?, ...)`.
+*/
+
+// containerIDFromURL gets the ID of a container from its URL.
+var containerIDFromURL = fmt.Sprintf(`SELECT ?, instances.id FROM instances JOIN projects ON instances.project_id = projects.id WHERE projects.name = ? AND '' = ? AND instances.name = ? AND instances.type = %d`, instancetype.Container)
```

Comment:
Its pretty hard to review these queries without knowing that the bindings are.

For example, this looks weird, but not sure if its correct?
`'' = ?`

Reply:
Could you explain this bit further?

> An identifier for the query being executed, this is directly returned so that we can match the result of the query to the URL (see PopulateEntityReferencesFromURLs below).

I'm assuming this is related to the `SELECT ?,` bit at the start of each query.
This seems odd to send a piece of data to the DB only to get it back and literally use it?